sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -82,7 +82,9 @@ case class GpuCreateNamedStruct(children: Seq[Expression]) extends GpuExpression
   }.toList.unzip
 
   private lazy val names = nameExprs.map {
-    case g: GpuExpression => g.columnarEval(null)
+    case gl: GpuLiteral => gl.value
```

Comment:
The reason for this change is not clear. Could we get a comment to explain why this is happening so no one accidentally changes it back.

Reply:
Done